internal/plugin.go

```diff
@@ -0,0 +1,660 @@
+	// replayer. Implementers should return an error or invoke next.
+	ReplayWorkflow(
+		ctx context.Context,
+		options WorkerPluginReplayWorkflowOptions,
+		next func(context.Context, WorkerPluginReplayWorkflowOptions) error,
+	) error
+
+	// Plugins must embed [go.temporal.io/sdk/worker.PluginBase].
+	mustEmbedWorkerPluginBase()
+}
+
+// WorkerPluginBase must be embedded into worker plugin implementations.
+//
+// Exposed as: [go.temporal.io/sdk/worker.PluginBase]
+//
+// NOTE: Experimental
+type WorkerPluginBase struct{}
+
+var _ WorkerPlugin = struct {
+	pluginNamePanicForTypeChecking
+	WorkerPluginBase
+}{}
+
+// WorkerPluginConfigureWorkerOptions are options for ConfigureWorker on a
+// worker plugin.
+//
+// Exposed as: [go.temporal.io/sdk/worker.PluginConfigureWorkerOptions]
+//
+// NOTE: Experimental
+type WorkerPluginConfigureWorkerOptions struct {
+	// WorkerInstanceKey is the unique, immutable instance key for this worker.
+	WorkerInstanceKey string
+
+	// TaskQueue is the immutable task queue for this worker.
+	TaskQueue string
+
+	// WorkerOptions are the set of mutable options that can be adjusted by
+	// plugins.
+	WorkerOptions *WorkerOptions
+
+	// WorkerRegistryOptions are the set of callbacks that can be adjusted by
+	// plugins. If adjusting a callback that is already set, implementers may
+	// want to take care to invoke the existing callback inside their own.
+	WorkerRegistryOptions *WorkerPluginConfigureWorkerRegistryOptions
+}
+
+// WorkerPluginConfigureWorkerRegistryOptions are the set of callbacks that can
+// be adjusted by plugins when configuring workers. If adjusting a callback that
+// is already set, implementers may want to take care to invoke the existing
+// callback inside their own.
+//
+// Exposed as: [go.temporal.io/sdk/worker.PluginConfigureWorkerRegistryOptions]
+//
+// NOTE: Experimental
+type WorkerPluginConfigureWorkerRegistryOptions struct {
+	OnRegisterWorkflow        func(any, RegisterWorkflowOptions)
+	OnRegisterDynamicWorkflow func(any, DynamicRegisterWorkflowOptions)
+	OnRegisterActivity        func(any, RegisterActivityOptions)
+	OnRegisterDynamicActivity func(any, DynamicRegisterActivityOptions)
+	OnRegisterNexusService    func(*nexus.Service)
+}
+
+// WorkerPluginStartWorkerOptions are options for StartWorker on a worker
+// plugin.
+//
+// Exposed as: [go.temporal.io/sdk/worker.PluginStartWorkerOptions]
+//
+// NOTE: Experimental
+type WorkerPluginStartWorkerOptions struct {
+	// WorkerInstanceKey is the unique, immutable instance key for this worker.
+	WorkerInstanceKey string
+
+	// WorkerRegistry is the worker registry plugins can use to register items
+	// with the worker. Implementers should usually not mutate this before
+	// passing to "next", but instead add register callbacks in ConfigureWorker
+	// if needed. Calls to this registry to invoke the OnX callbacks set in
+	// ConfigureWorker.
+	WorkerRegistry interface {
```

Comment:
What is the difference between this interface and `WorkerPluginConfigureWorkerRegistryOptions`?